tests/integration/ha_tests/test_scaling.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+import logging
+from asyncio import gather
+
+import pytest
+from pytest_operator.plugin import OpsTest
+
+from .. import markers
+from ..helpers import (
+    CHARM_BASE,
+    DATABASE_APP_NAME,
+)
+from .conftest import APPLICATION_NAME
+from .helpers import (
+    app_name,
+    are_writes_increasing,
+    check_writes,
+    get_cluster_roles,
+    get_primary,
+    start_continuous_writes,
+)
+
+logger = logging.getLogger(__name__)
+
+charm = None
+
+
+@pytest.mark.group(1)
+@markers.juju3
```

Comment:
Only tested on juju 3 for the moment. Should be able to reuse the wrapper around force removal to enable the tests on juju 2